tickit/adapters/zmqadapter.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+import logging
+from dataclasses import dataclass
+from typing import AsyncIterable, Awaitable, Callable, List
+
+import aiozmq
+import zmq
+
+from tickit.core.adapter import ConfigurableAdapter
+from tickit.core.device import ConfigurableDevice
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ZeroMQStream(ConfigurableDevice):
+    """A ZeroMQ data stream."""
+
+    async def update(self) -> AsyncIterable[int]:
```

Comment:
This breaks the `Device` protocol so means it's not a device.
I would recommend making a version of `tickit.devices.source.Source` that increments `self.value` every update and adding a getter rather than an async iterator for the adapter.

Reply:
done this in #29

Reply:
That's okay, the point is to make that optional/up to the implementer of the device